src/main/java/net/mcreator/element/types/Biome.java

```diff
@@ -92,33 +80,55 @@
 	public List<String> biomeDictionaryTypes;
 	public List<String> defaultFeatures;
 
+	public List<SpawnTree> spawnTrees;
 	public List<SpawnEntry> spawnEntries;
 
+	public boolean customTree;
+	public int minHeight;
+	public double extraChance;
+	public int extraCount;
+	public boolean spawnVines;
+	public MItemBlock treeVines;
+	public MItemBlock treeStem;
+	public MItemBlock treeBranch;
+	public MItemBlock treeFruits;
+	public int treesPerChunk;
+
 	private Biome() {
 		this(null);
 	}
 
 	public Biome(ModElement element) {
 		super(element);
 
-		// RESTORE CONSTANTS (FOR GSON)
-		TREES_VANILLA = 0;
-		TREES_CUSTOM = 1;
-
 		// DEFAULT VALUES
 		name = "";
 		spawnStronghold = true;
 		spawnMineshaft = true;
 		spawnPillagerOutpost = true;
-		vanillaTreeType = "Default";
 		villageType = "none";
 		oceanRuinType = "NONE";
 		biomeCategory = "NONE";
 		biomeDictionaryTypes = new ArrayList<>();
+		spawnTrees = new ArrayList<>();
 		spawnEntries = new ArrayList<>();
 		defaultFeatures = new ArrayList<>();
 	}
 
+	public static class SpawnTree {

```

Comment:
Refactor (Shift + F6 in IDEA to rename all references) to `TreeSpawn`, SpawnTree is a bit strange word order.

Rename field spawnTrees to treeSpawns too.